pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -74,7 +76,7 @@ func NewDiscoveryServer(grpcServer *grpc.Server, env model.Environment) *Discove
 	out := &DiscoveryServer{
 		GrpcServer:      grpcServer,
 		env:             env,
-		ConfigGenerator: core.NewConfigGenerator(),
+		ConfigGenerator: v1alpha3.NewConfigGenerator(plugin.NewPlugins()),
```

Comment:
The reason I wanted to use core instead of v1alpha3 was so that we could transparently swap from v1alpha3 to v1alpha4 or what not without having to worry about the underlying implementation. Sort of like the factory method.
Directly using v1alpha3 creates specific dependencies throughout the code base and people tend to use functions directly from v1alpha3 instead of those exposed by core alone. 

This will also make it easy for other folks to simply consume the networking/core library without having ton consume our gRPC server, or consume our gRPC server but have a different implementation of the routing APIs.

Reply:
Same applies to directly instantiating plugins as well. I don't knwo the go way of doing it, but you get the idea..

Reply:
Instantiating new plugins that we use dynamically is still a problem with the current setup. I'll have to think about how we make it easier to add plugins (other than what's required today, which is adding it to the list in `registry/registry.go`).